homeassistant/components/asuswrt/bridge.py

```diff
@@ -236,38 +260,22 @@ async def _get_available_temperature_sensors(self) -> list[str]:
         availability = await self._api.async_find_temperature_commands()
         return [SENSORS_TEMPERATURES[i] for i in range(3) if availability[i]]
 
-    async def _get_bytes(self) -> dict[str, Any]:
+    @handle_errors_and_zip((IndexError, OSError, ValueError), SENSORS_BYTES)
+    async def _get_bytes(self) -> Any:
         """Fetch byte information from the router."""
-        try:
-            datas = await self._api.async_get_bytes_total()
-        except (IndexError, OSError, ValueError) as exc:
-            raise UpdateFailed(exc) from exc
-
-        return _get_dict(SENSORS_BYTES, datas)
+        return await self._api.async_get_bytes_total()
 
-    async def _get_rates(self) -> dict[str, Any]:
+    @handle_errors_and_zip((IndexError, OSError, ValueError), SENSORS_RATES)
+    async def _get_rates(self) -> Any:
         """Fetch rates information from the router."""
-        try:
-            rates = await self._api.async_get_current_transfer_rates()
-        except (IndexError, OSError, ValueError) as exc:
-            raise UpdateFailed(exc) from exc
-
-        return _get_dict(SENSORS_RATES, rates)
+        return await self._api.async_get_current_transfer_rates()
 
-    async def _get_load_avg(self) -> dict[str, Any]:
+    @handle_errors_and_zip((IndexError, OSError, ValueError), SENSORS_LOAD_AVG)
+    async def _get_load_avg(self) -> Any:
         """Fetch load average information from the router."""
-        try:
-            avg = await self._api.async_get_loadavg()
-        except (IndexError, OSError, ValueError) as exc:
-            raise UpdateFailed(exc) from exc
+        return await self._api.async_get_loadavg()
 
-        return _get_dict(SENSORS_LOAD_AVG, avg)
-
-    async def _get_temperatures(self) -> dict[str, Any]:
+    @handle_errors_and_zip((OSError, ValueError), None)
+    async def _get_temperatures(self) -> Any:
```

Comment:
```suggestion
    async def _get_temperatures(self) -> dict[str, Any]:
```

Reply:
Here I always receive this error from mypy:

`Returning Any from function declared to return "dict[str, Any]"  [no-any-return]`

and I don't know how how to fix. Should I cast the function?

Reply:
The library is not typed so mypy assumes Any as the return type.
We leave `Any` here for now and if you want, you can add type hints to the library and so we can change the Any in the future